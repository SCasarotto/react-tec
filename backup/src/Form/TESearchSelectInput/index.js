import TESearchSelectInput from './TESearchSelectInput'

export default TESearchSelectInput
