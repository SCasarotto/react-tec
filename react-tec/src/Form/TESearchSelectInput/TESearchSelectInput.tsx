import React from 'react'
import PropTypes from 'prop-types'
import { PowerSelect, PowerSelectMultiple } from 'react-power-select'
// import 'react-power-select/dist/react-power-select.css'

//NOTE:
//
// - At this moment there is no way to style the dropdown component with styled components.
//

const TESearchSelectInput = (props) => {
	const { type = 'single', searchEnabled = true, value, className = '', ...rest } = props

	if (type === 'single') {
		return (
			<PowerSelect
				selected={value}
				className={`TESearchSelectInput ${className}`}
				{...rest}
			/>
		)
	}
	if (type === 'multi') {
		return (
			<PowerSelectMultiple
				selected={value}
				className={`TESearchSelectInput ${className}`}
				{...rest}
			/>
		)
	}

	return <PowerSelect selected={value} className={`TESearchSelectInput ${className}`} {...rest} />
}

TESearchSelectInput.propTypes = {
	type: PropTypes.string,
}

export default TESearchSelectInput
