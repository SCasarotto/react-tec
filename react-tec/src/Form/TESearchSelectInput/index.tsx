import React from 'react'
import { PowerSelect, PowerSelectMultiple } from 'react-power-select'
// import 'react-power-select/dist/react-power-select.css'

//NOTE:
//
// - At this moment there is no way to style the dropdown component with styled components.
//

//TODO: Return to this.
export const TESearchSelectInput: React.FC<any> = (props) => {
	const {
		type = 'single',
		searchEnabled = true,
		value,
		className = '',
		...rest
	} = props

	if (type === 'single') {
		return (
			<PowerSelect
				selected={value}
				className={`TESearchSelectInput ${className}`}
				searchEnabled={searchEnabled}
				{...rest}
			/>
		)
	}
	if (type === 'multi') {
		return (
			<PowerSelectMultiple
				selected={value}
				className={`TESearchSelectInput ${className}`}
				searchEnabled={searchEnabled}
				{...rest}
			/>
		)
	}

	return (
		<PowerSelect
			selected={value}
			className={`TESearchSelectInput ${className}`}
			{...rest}
		/>
	)
}
