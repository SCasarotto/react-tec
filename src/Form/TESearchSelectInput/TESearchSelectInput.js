//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'
import { PowerSelect, PowerSelectMultiple } from 'react-power-select'
// import 'react-power-select/dist/react-power-select.css'

// import { } from './styledComponents'

//NOTE:
//
// - At this moment there is no way to style the dropdown component with styled components.
//

const TESearchSelectInput = (props) => {
    const { type, value, ...rest } = props

    if (type === 'single') {
        return <PowerSelect selected={value} {...rest} />
    }
    if (type === 'multi') {
        return <PowerSelectMultiple selected={value} {...rest} />
    }

    return <PowerSelect selected={value} {...rest} />
}

TESearchSelectInput.propTypes = {
    type: PropTypes.string,
}

TESearchSelectInput.defaultProps = {
    type: 'single',
    searchEnabled: true,
}

export default TESearchSelectInput
