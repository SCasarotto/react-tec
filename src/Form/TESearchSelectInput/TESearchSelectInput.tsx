import React from 'react'
import { PowerSelect, PowerSelectMultiple } from 'react-power-select'
// import 'react-power-select/dist/react-power-select.css'

//NOTE:
//
// - At this moment there is no way to style the dropdown component with styled components.
//

//TODO: Return to this.
export const TESearchSelectInput = React.forwardRef<any, any>((props, ref) => {
	const { type = 'single', searchEnabled = true, value, className = '', ...rest } = props

	if (type === 'single') {
		return (
			<PowerSelect
				selected={value}
				className={`TESearchSelectInput ${className}`}
				searchEnabled={searchEnabled}
				ref={ref}
				{...rest}
			/>
		)
	}
	if (type === 'multi') {
		return (
			<PowerSelectMultiple
				selected={value}
				className={`TESearchSelectInput ${className}`}
				searchEnabled={searchEnabled}
				ref={ref}
				{...rest}
			/>
		)
	}

	return <PowerSelect selected={value} className={`TESearchSelectInput ${className}`} {...rest} />
})
