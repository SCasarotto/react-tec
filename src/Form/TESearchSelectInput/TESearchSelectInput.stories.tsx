import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TESearchSelectInput } from './TESearchSelectInput'

export default {
	title: 'Form/TESearchSelectInput',
	component: TESearchSelectInput,
	argTypes: {},
} as Meta

const Template: Story<any> = (args) => <TESearchSelectInput {...args} />

export const Default = Template.bind({})
Default.args = {}
