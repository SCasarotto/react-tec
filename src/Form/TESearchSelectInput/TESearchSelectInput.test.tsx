import React from 'react'
import { render } from '@testing-library/react'
import { TESearchSelectInput } from './TESearchSelectInput'

describe('TESearchSelectInput', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(<TESearchSelectInput onChange={() => {}} />)
		expect(asFragment()).toMatchSnapshot()
	})
})
