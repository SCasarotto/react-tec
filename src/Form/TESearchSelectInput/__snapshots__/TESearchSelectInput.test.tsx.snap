// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESearchSelectInput matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="PowerSelect TESearchSelectInput "
    tabindex="0"
  >
    <div
      class="PowerSelect__Trigger PowerSelect__Trigger--empty"
    >
      <div
        class="PowerSelect__TriggerLabel"
      >
        <span
          class="PowerSelect__Placeholder"
        />
      </div>
      <span
        class="PowerSelect__Clear"
      />
      <span
        class="PowerSelect__TriggerStatus"
      />
    </div>
  </div>
</DocumentFragment>
`;
